src/field/field3d.cxx

```diff
@@ -401,6 +399,9 @@ Field3D & Field3D::operator=(const Field2D &rhs) {
 void Field3D::operator=(const FieldPerp &rhs) {
   ASSERT1(rhs.isAllocated());
```

Comment:
Can this be removed?

Reply:
Yes, sorry, I meant to replace all the plain `isAllocated` checks with `checkData`